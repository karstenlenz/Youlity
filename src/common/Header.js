import React from 'react'
import { Link, useHistory } from 'react-router-dom'
import styled from 'styled-components/macro'
import IconButton from './IconButton'
import { ReactComponent as Logo } from '../img/logo.svg'
import { ReactComponent as BackIcon } from '../img/back.svg'
import { ReactComponent as HomeIcon } from '../img/home.svg'

export default function Header() {
  const history = useHistory()

  return (
    <HeaderContainer>
      <IconButton onClick={() => history.goBack()}>
        <BackIcon title="zurÃ¼ck" />
      </IconButton>
      <Link to="/">
        <Logo title="Youlity logo" />
      </Link>
      <IconButton onClick={() => history.push('/')}>
        <HomeIcon title="zur Homepage" />
      </IconButton>
    </HeaderContainer>
  )
}

const HeaderContainer = styled.header`
  display: grid;
  grid-template-columns: 1fr auto 1fr;
  height: 60px;
  width: 100%;
  padding: 10px 15px;
  box-shadow: var(--primary-shadow);
  button:nth-child(3) {
    justify-self: end;
  }
`
