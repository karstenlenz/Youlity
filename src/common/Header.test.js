import { render } from '@testing-library/react'
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import Header from './Header'

describe('The Header component', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Router>
        <Header>Header content lorem ipsum</Header>
      </Router>
    )
    expect(container).toMatchSnapshot()
  })
})
