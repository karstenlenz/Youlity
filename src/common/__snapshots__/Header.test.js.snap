// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header component renders correctly 1`] = `
<div>
  <header
    class="Header__HeaderContainer-sc-15lxbtt-0 lkLVpF"
  >
    <button
      class="IconButton__ButtonContainer-sc-47n1yg-0 jCeVKT"
    >
      <img
        alt="zurÃ¼ck"
        src="/img/back.svg"
      />
    </button>
    <a
      href="/"
    >
      <img
        alt="Youlity logo"
        src="/img/logo.svg"
      />
    </a>
    <button
      class="IconButton__ButtonContainer-sc-47n1yg-0 jCeVKT"
    >
      <img
        alt="zur Homepage"
        src="/img/home.svg"
      />
    </button>
  </header>
</div>
`;
